WalletWasabi.Fluent/ViewModels/MainViewModel.cs

```diff
@@ -76,57 +64,26 @@ public MainViewModel()
 			.ObserveOn(RxApp.MainThreadScheduler)
 			.Subscribe(state => Services.UiConfig.WindowState = state.ToString());
 
-		this.WhenAnyValue(
-				x => x.DialogScreen!.IsDialogOpen,
-				x => x.FullScreen!.IsDialogOpen,
-				x => x.CompactDialogScreen!.IsDialogOpen)
+		IsMainContentEnabled = this.WhenAnyValue(
+				x => x.DialogScreen.IsDialogOpen,
+				x => x.FullScreen.IsDialogOpen,
+				x => x.CompactDialogScreen.IsDialogOpen)
 			.ObserveOn(RxApp.MainThreadScheduler)
-			.Subscribe(tup =>
-			{
-				var (dialogScreenIsOpen, fullScreenIsOpen, compactDialogScreenIsOpen) = tup;
-
-				IsMainContentEnabled = !(dialogScreenIsOpen || fullScreenIsOpen || compactDialogScreenIsOpen);
-			});
+			.Select(tup => !(tup.Item1 || tup.Item2 || tup.Item3));
 
 		this.WhenAnyValue(
 				x => x.DialogScreen.CurrentPage,
 				x => x.CompactDialogScreen.CurrentPage,
 				x => x.FullScreen.CurrentPage,
 				x => x.MainScreen.CurrentPage)
-			.ObserveOn(RxApp.MainThreadScheduler)
-			.Subscribe(tup =>
+			.Select(tup =>
 			{
 				var (dialog, compactDialog, fullscreenDialog, mainsScreen) = tup;
-
-				/*
-				 * Order is important.
-				 * Always the topmost content will be the active one.
-				 */
-
-				if (compactDialog is { })
-				{
-					compactDialog.SetActive();
-					return;
-				}
-
-				if (dialog is { })
-				{
-					dialog.SetActive();
-					return;
-				}
-
-				if (fullscreenDialog is { })
-				{
-					fullscreenDialog.SetActive();
-					return;
-				}
-
-				if (mainsScreen is { })
-				{
-					mainsScreen.SetActive();
-					return;
-				}
-			});
+				return compactDialog ?? dialog ?? fullscreenDialog ?? mainsScreen; // Order is important. Always the topmost content will be the active one.
+			})
+			.WhereNotNull()
+			.ObserveOn(RxApp.MainThreadScheduler)
+			.Subscribe(page => page.SetActive());
```

Comment:
This should be equivalent, but with slightly better semantics and more compact.
```
		this.WhenAnyValue(
				x => x.DialogScreen.CurrentPage,
				x => x.CompactDialogScreen.CurrentPage,
				x => x.FullScreen.CurrentPage,
				x => x.MainScreen.CurrentPage, (dialog, compactDialog, fullscreenDialog, mainScreen) => compactDialog ?? dialog ?? fullscreenDialog ?? mainScreen)
			.WhereNotNull()
			.ObserveOn(RxApp.MainThreadScheduler)
			.Do(page => page.SetActive())
			.Subscribe();
```

Remember that `Subscribe` should not contain "interesting" code :)

Reply:
fixed, thanks